engine/action_transaction.go

```diff
@@ -360,7 +361,11 @@ func (c *Client) HandleTxCallback(ctx context.Context, callbackResp *chainmodels
 	}
 
 	tx.BlockHash = callbackResp.BlockHash
-	tx.BlockHeight = uint64(callbackResp.BlockHeight)
+	if callbackResp.BlockHeight < 0 {
+		return fmt.Errorf("invalid block height: %d", callbackResp.BlockHeight)
+	}
+	// Suppress gosec G115 warning as the conversion is safe due to prior negative check
+	tx.BlockHeight = uint64(callbackResp.BlockHeight) //nolint:gosec
```

Comment:
Similarly, we could introduce some util function e.g. `func ConvertIntToUInt64(val int) (uint64, error)`.

Or even a second version:
```golang
func ShouldConvertIntToUInt64(value int) uint64 {
    converted, err := ConvertIntToUInt64(value)
    if err != nil {
        panic(err)
    }
    return converted
}
```